src/cephadm/cephadmlib/daemons/smb.py

```diff
@@ -432,7 +382,7 @@ def __init__(self, ctx: CephadmContext, ident: DaemonIdentity):
         self._raw_configs: Dict[str, Any] = context_getters.fetch_configs(ctx)
         self._config_keyring = context_getters.get_config_and_keyring(ctx)
         self._cached_layout: Optional[ContainerLayout] = None
-        self._rank_info = context_getters.fetch_rank_info(ctx)
+        self._rank_info = context_getters.fetch_rank_info(ctx) or (-1, -1)
```

Comment:
Was there some additional issue related to this not getting populated or is this necessitated by the use of the dataclass in some way I'm missing?

Reply:
Yes, the latter. The previous code assigned a type of `Optional[Tuple[int, int]]` to self._rank_info (implicitly). Then in the other function it created the Config and *if* self._rank_info was not None it updated the config object with tuple unpacking. Since I made the dataclass frozen, we can't update it. So I found it easier to make sure self._rank_info is always a tuple and the tuple is trivially unpacked right before the Config object is created.